typescript/polyglotpkg/src/packager.ts

```diff
@@ -11,84 +11,73 @@ import createLogger from './lib/logger';
-        for (var i = 0; i < projectActions.length; i++) {
-            const actionType = projectActions[i].actionType;
-            const actionRuntime = projectActions[i].actionRuntime;
-
-            if (actionType === ActionType.UNKNOWN || actionRuntime === ActionRuntime.UNKNOWN) {
-                throw new Error(`Unsupported action type or runtime: ${actionType} ${actionRuntime}`);
-            }
-
-            this.logger.info(`Packaging ${actionType} ${actionRuntime} action from ${projectActions[i].src}...`);
-
-            let strategy: IStrategy;
-
-            switch (actionRuntime) {
-                case ActionRuntime.ABX_NODEJS:
-                case ActionRuntime.VRO_NODEJS_12:
-                case ActionRuntime.VRO_NODEJS_14:
-                    strategy = new NodejsStrategy(this.logger, projectActions[i], (e: Events) => this.emit(e));
-                    await strategy.packageProject();
-                    break;
-                case ActionRuntime.ABX_PYTHON:
-                case ActionRuntime.VRO_PYTHON_310:
-                case ActionRuntime.VRO_PYTHON_37:
-                    strategy = new PythonStrategy(this.logger, projectActions[i], (e: Events) => this.emit(e));
-                    await strategy.packageProject();
-                    break;
-                case ActionRuntime.ABX_POWERSHELL:
-                case ActionRuntime.VRO_POWERCLI_11_PS_62:
-                case ActionRuntime.VRO_POWERCLI_12_PS_71:
-                    strategy = new PowershellStrategy(this.logger, projectActions[i], (e: Events) => this.emit(e));
-                    await strategy.packageProject();
-                    break;
-                default:
-                    throw new Error(`Action runtime ${actionRuntime} is not yet supported`);
-            }
-
-            // Prepare input files for vRO packaging
-            if (!this.options.skipVro && actionType === ActionType.VRO) {
-                const tree = new VroTree(this.logger, projectActions[i]);
-                await tree.createTree();
-            }
-
-            // Prepare input files for ABX packaging
-            if (actionType === ActionType.ABX) {
-                await createPackageJsonForABX(projectActions[i], projectActions[i].mixed);
-            }
-
-        }
-
-
-    }
+	private readonly logger: Logger;
+
+	constructor(private readonly options: PackagerOptions) {
+		super();
+		this.logger = createLogger(false, options.outputStream);
+	}
+
+	async packageProject() {
+		// Collect list of actions included in the project
+		const projectActions = await getProjectActions(this.options, <ActionType>this.options.env);
+
+		/*-
+		 * #%L
+		 * polyglotpkg
+		 * %%
+		 * Copyright (C) 2023 - 2024 VMware
+		 * %%
+		 * Build Tools for VMware Aria
+		 * Copyright 2023 VMware, Inc.
+		 *
+		 * This product is licensed to you under the BSD-2 license (the "License"). You may not use this product except in compliance with the BSD-2 License.
```

Comment:
Love the licensing placement...this is the same issue that we discussed in here right?
https://github.com/vmware/build-tools-for-vmware-aria/issues/381

Reply:
Yes... It's driving me insane... it also messes up my IDE whenever it happens...